cms/djangoapps/modulestore_migrator/api/read_api.py

```diff
@@ -242,3 +252,113 @@ def _block_migration_success(
+            "total_components": 0,
+            "sections": 0,
+            "subsections": 0,
+            "units": 0,
+        }
+    ```
+
+    List of states:
+    - 'success': The migration can be carried out in its entirety
+    - 'partial': The migration will be partial, because there are unsupported blocks.
+    - 'block_limit_reached': The migration cannot be performed because the block limit per library has been reached.
+
+    TODO: For now, the repeat_handling_strategy is not taken into account. This can be taken into
+    account for a more advanced summary.
+    """
+    # Get all containers and components from the source key
+    blocks = get_all_blocks_from_context(source_key, ["block_type", "block_id"])
+
+    unsupported_blocks = []
+    total_blocks = 0
+    total_components = 0
+    sections = 0
+    subsections = 0
+    units = 0
+    blocks_limit = settings.MAX_BLOCKS_PER_CONTENT_LIBRARY
+
+    # Builds the summary: counts every container and verify if each component can be added to the library 
+    for block in blocks:
+        block_type = block["block_type"]
+        block_id = block["block_id"]
+        total_blocks += 1
+        if block_type not in ['chapter', 'sequential', 'vertical']:
+            total_components += 1
+            try:
+                validate_can_add_block_to_library(
+                    target_key,
+                    block_type,
+                    block_id,
+                )
+            except BlockLimitReachedError:
+                return {
+                    "state": "block_limit_reached",
+                    "unsupported_blocks": 0,
+                    "unsupported_percentage": 0,
+                    "blocks_limit": blocks_limit,
+                    "total_blocks": 0,
+                    "total_components": 0,
+                    "sections": 0,
+                    "subsections": 0,
+                    "units": 0,
+                }
+            except (IncompatibleTypesError, PluginMissingError):
+                unsupported_blocks.append(block["usage_key"])
+            except LibraryBlockAlreadyExists:
+                # Skip this validation, The block may be repeated in the library, but that's not a bad thing.
+                pass
+        elif block_type == "chapter":
+            sections += 1
+        elif block_type == "sequential":
+            subsections += 1
+        elif block_type == "vertical":
+            units += 1
+
+    # Gets the count of children of unsupported blocks
+    quoted_keys = ','.join(f'"{key}"' for key in unsupported_blocks)
+    unsupportedBlocksChildren = fetch_block_types(
+        [
+            f'context_key = "{source_key}"',
+            f'breadcrumbs.usage_key IN [{quoted_keys}]'
+        ],
+    )
+    # Final unsupported blocks count
+    unsupported_blocks_count = len(unsupported_blocks) + unsupportedBlocksChildren["estimatedTotalHits"]
+    unsupported_percentage = (unsupported_blocks_count / total_blocks) * 100
```

Comment:
@ChrisChV As per latest requirements, we need to ignore children counts from the total counts and percentage. ~So we can skip this part.~ You'll still need to fetch this data to subtract the children count from total block counts.

See https://github.com/openedx/frontend-app-authoring/issues/2525#issuecomment-3684097851 and the related PR: https://github.com/openedx/frontend-app-authoring/pull/2774